src/Engine/GameEngine.cs

```diff
@@ -81,5 +88,55 @@ public void OnError(Exception error)
         {
             throw new NotImplementedException();
         }
+
+        #region IDisposable Implementation
+
+        private bool _disposedValue = false;
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposedValue)
+            {
+                if (disposing)
```

Comment:
Often easier to remove a nested `if` by inverting a return. i.e. `if (!disposing) return;`